### DISCUSSION TITLE
Why does joining increase vertex count?

### DISCUSSION THREAD

**Discussion by stevenpaulhoward** [donmccurdy/glTF-Transform#852] [General]
Under the description of the [join function](https://gltf-transform.donmccurdy.com/functions.html#join) it is stated:
> NOTE: In a Scene that heavily reuses the same Mesh data, joining may increase vertex count. Consider alternatives, like instancing with EXTMeshGPUInstancing

I've noticed this while using the join function out of the package and when joining meshes in Blender. I'm curious what is happening to cause an increase in vertex count?

**My crude attempt at understanding what is going on, likely very wrong and incorrect**: If a scene is reusing the same Mesh data, then that means multiple meshes in a scene are comprised of primitives referencing the same accessor where index buffer data is stored? and then TRS is being applied at the mesh / node level? However, when you run join, you are joining primitives, and now have to create new data representing where those referenced vertex attributes are actually positioned in the scene. That new data is appended to the index buffer and you thus have a longer buffer with much more data, causing larger file size as well? But I guess that still doesn't answer my original question of why that causes an increase in vertex count.... Hmm. 

Trying to wrap my head around this 😅 any clarity anyone can provide and/or share resources that would better depict what is going on would be greatly appreciated! 🙏

**Comment by donmccurdy** [comment]
The example you gave is one of several similar things that could happen here, yes. glTF can also reuse the same mesh on many nodes without the indirection of sharing accessors - the mesh itself can be reused. Joining results in fewer, larger meshes.

The number also depends on how your software is adding up the vertex count, particularly those reused meshes - the GPU is not necessarily doing more work, and will often be doing less. But if you have a lot of reuse (say, hundreds of the same mesh) then instancing would be more appropriate.

**Reply by stevenpaulhoward** [reply]
Gotcha, thank you for the elaboration 🙌

**Comment by Ruth678** [comment]
Hy,
Nice it is very interesting.
